### ISSUE TITLE
rdfimport: better error reporting when external data source already exists

### ISSUE THREAD

**Issue by koeaw** [acdh-oeaw/apis-core-rdf#809]
When trying to merge a local entity with data from an external source _whose URL already exists_, an `IntegrityError` error is thrown.

For end users, this currently manifests in a non-descript 500 server error which doesn't tell them what went wrong.

**Comment by b1rger** [comment]
Is there more information about this IntegrityError? Maybe where it is thrown? The `apis_core.utils.helpers.create_object_from_uri` method first looks up the URI and if it exists it should return the linked object. Only if the URI is not found in the DB, it tries to run the importer. So it would be great to know where the error occurs

**Comment by koeaw** [comment]
I might have got/let myself get confused about the source of this error after briefly discussing it with @sennierer (?!) – my original suspicion was the history app (see also mention in ProsNet channel on MM).

The error message reads:

```
IntegrityError at /apis/entities/entity/person/18799/merge/

null value in column "history_date" of relation "apis_relations_versiontemptriple" violates not-null constraint
DETAIL:  Failing row contains (22502, 22502, null, f, null, null, null, null, null, null, null, null, , null, null, 5, null, null, -, null, 18799, 18, 18798).
```

The error occurs when trying to merge my local version of [this object](https://frischmuth-dev.acdh-dev.oeaw.ac.at/apis/entities/entity/person/18799/edit/) with [this DNB entry](https://d-nb.info/gnd/118527576).

`history_date` seems to point to the history app being at fault here. Reason why we thought the problem might be related to an existing `Uri` is that I've been able to import from the same source to the same entity type (DNB -> `Person`) without problems for an object which didn't already have a DNB `Uri` attached.

**Comment by babslgam** [comment]
> I might have got/let myself get confused about the source of this error after briefly discussing it with @sennierer (?!) – my original suspicion was the history app (see also mention in ProsNet channel on MM).
> 
> The error message reads:
> 
> ```
> IntegrityError at /apis/entities/entity/person/18799/merge/
> 
> null value in column "history_date" of relation "apis_relations_versiontemptriple" violates not-null constraint
> DETAIL:  Failing row contains (22502, 22502, null, f, null, null, null, null, null, null, null, null, , null, null, 5, null, null, -, null, 18799, 18, 18798).
> ```
> 
> The error occurs when trying to merge my local version of [this object](https://frischmuth-dev.acdh-dev.oeaw.ac.at/apis/entities/entity/person/18799/edit/) with [this DNB entry](https://d-nb.info/gnd/118527576).
> 
> `history_date` seems to point to the history app being at fault here. Reason why we thought the problem might be related to an existing `Uri` is that I've been able to import from the same source to the same entity type (DNB -> `Person`) without problems for an object which didn't already have a DNB `Uri` attached.

@koeaw can you confirm that you get the same error when trying to delete an object?

**Comment by koeaw** [comment]
> @koeaw can you confirm that you get the same error when trying to delete an object?

I haven't yet tried that, but can do so.

**Comment by koeaw** [comment]
Right, I can't delete the local version of the object I linked to above. @babslgam is this something you came across randomly or is this related to existing reports of deletions not working or similar?

**Comment by b1rger** [comment]
ping?

**Comment by koeaw** [comment]
I found another object to test this with – in this case, the two local objects – the one into which I wanted to merge the DNB data and the one which already had the Uri added – got merged together.

Which I guess makes sense? But hmm, could potentially also lead to accidental incorrect merges.

Example scenario:  
We currently have what may or may not be duplicates of places, like Wien or Salzburg. It's possible these are meant to exist in parallel, one object for the city, one of for the federal state (unclear at this point). If this were the case, the team would ideally add more (meta) information for better differentiation, which could easily be done by merging the "other" DNB data into those objects. But I can see them selecting the wrong DNB source and thus accidentally merging two local objects which shouldn't be merged.
